exasol/toolbox/util/dependencies/track_vulnerabilities.py

```diff
@@ -0,0 +1,43 @@
+from pydantic import (
+    BaseModel,
+    ConfigDict,
+)
+
+from exasol.toolbox.util.dependencies.audit import Vulnerability
+
+
+class ResolvedVulnerabilities(BaseModel):
+    model_config = ConfigDict(frozen=True, arbitrary_types_allowed=True)
+
+    previous_vulnerabilities: list[Vulnerability]
+    current_vulnerabilities: list[Vulnerability]
+
+    def _is_resolved(self, previous_vuln: Vulnerability):
+        """
+        Detects if a vulnerability has been resolved.
+
+        A vulnerability is said to be resolved when it cannot be found
+        in the `current_vulnerabilities`. In order to see if a vulnerability
+        is still present, its id and aliases are compared to values in the
+        `current_vulnerabilities`. While one could hope that the IDs
+        are the same, it's possible between pip-audit versions that
+        there are differences.
```

Comment:
I think, vulnerabilities can change their ID independently of pip-audit versions. I would hope this is also dealt with using the aliases.